server/encryption/localkey_handler.go

```diff
@@ -0,0 +1,217 @@
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"io"
+	"os"
+
+	"github.com/google/tink/go/kwp/subtle"
+)
+
+const (
+	// DataKeyLength provides the length for data key in bytes.
+	// It is recommended to use an authentication key with 32 or 64 bytes.
+	// The data key length must be either
+	// 16, 24, or 32 bytes to select AES-128, AES-192, or AES-256 modes.
+	DataKeyLength int = 32
+)
+
+var (
+	masterKeyVarName = "LOCAL_MASTER_KEY"
+)
+
+// LocalEncryptionHandler provides functionalities to load secret key
+// from environment variables
+type LocalEncryptionHandler struct {
+	defaultDKS  []byte
+	keywrapper  *subtle.KWP
+	blockCypher *cipher.AEAD
+}
+
+// NewLocalEncriptionHandler generates a new instance of LocalEncryptionHandler
+func NewLocalEncriptionHandler() (*LocalEncryptionHandler, error) {
+	localEncryptionHandler := LocalEncryptionHandler{}
+
+	// Init key wrapper
+
+	masterKeyStr := os.Getenv(masterKeyVarName)
+	masterKey := []byte(masterKeyStr)
+	kwp, err := subtle.NewKWP(masterKey)
+
+	if err != nil {
+		return nil, err
+	}
+	localEncryptionHandler.keywrapper = kwp
+
+	return &localEncryptionHandler, nil
+}
+
+// generateDKS retrieves the pre-configurated encryption key
+// from the environment variables.
```

Comment:
Wait, the function doesn't load anything from an env variable. 🤔

Reply:
nope, it generate a random data key. The comment will be rectified.